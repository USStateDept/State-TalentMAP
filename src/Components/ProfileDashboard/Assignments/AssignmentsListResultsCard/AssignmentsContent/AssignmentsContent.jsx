import { get } from 'lodash';
// import { Link } from 'react-router-dom';
import { NO_LANGUAGES, NO_POSITION_NUMBER, NO_POST, NO_SKILL } from 'Constants/SystemMessages';
import { POSITION_DETAILS } from 'Constants/PropTypes';
import { formatDate, getPostName } from '../../../../../utilities';
import StartEnd from '../../../PositionInformation/StartEnd';


const AssignmentsContent = ({ assignment }) => (
  <div className="usa-grid-full bid-content-container">
    <div className="bid-list-card-title-lg">
      <span className="bid-list-card-title-post">{get(assignment, 'position.title')} </span>
    </div>
    <div>
      <span className="usa-sr-only">Position number: </span>
      <span className="bid-list-card-title-post bid-list-card-title-lg">
        {
          get(assignment, 'position.position_number') ?
            `(${get(assignment, 'position.position_number')}) ` : NO_POSITION_NUMBER
        }
      </span>
      {/* <Link to={`/archived/${get(assignment, 'position.position_id')}`}>View Position</Link> */}
    </div>
    <div>
      <span className="bid-list-card-title-post">Location: </span>
      {getPostName(get(assignment, 'position.post', NO_POST))}
    </div>
    <div>
      <span className="bid-list-card-title-post">Skill: </span>
      {get(assignment, 'position.skill', NO_SKILL)}
    </div>
    <div>
      <span className="bid-list-card-title-post">Language: </span>
      {get(assignment, 'position.language', NO_LANGUAGES)}
    </div>
    <div>
      <span className="bid-list-card-title-post">Start date and End date: </span>
      <StartEnd
        start={formatDate(assignment.start_date)}
        end={formatDate(assignment.end_date)}
      />
    </div>
  </div>
);

AssignmentsContent.propTypes = {
  assignment: POSITION_DETAILS.isRequired,
};


export default AssignmentsContent;
