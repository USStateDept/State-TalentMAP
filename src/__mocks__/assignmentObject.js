const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
