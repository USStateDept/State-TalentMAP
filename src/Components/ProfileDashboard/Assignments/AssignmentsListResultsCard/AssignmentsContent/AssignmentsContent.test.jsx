import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AssignmentsContent from './AssignmentsContent';
import assignmentObject from '../../../../../__mocks__/assignmentObject';

describe('AssignmentsContentComponent', () => {
  const assignment = assignmentObject;
  it('is defined', () => {
    const wrapper = shallow(
      <AssignmentsContent
        assignment={assignment}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <AssignmentsContent
        assignment={assignment}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
