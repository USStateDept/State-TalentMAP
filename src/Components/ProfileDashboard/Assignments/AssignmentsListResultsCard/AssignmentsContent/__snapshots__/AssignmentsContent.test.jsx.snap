// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentsContentComponent matches snapshot 1`] = `
<div
  className="usa-grid-full bid-content-container"
>
  <div
    className="bid-list-card-title-lg"
  >
    <span
      className="bid-list-card-title-post"
    >
      PROJECT DIRECTOR
       
    </span>
  </div>
  <div>
    <span
      className="usa-sr-only"
    >
      Position number: 
    </span>
    <span
      className="bid-list-card-title-post bid-list-card-title-lg"
    >
      (58348065) 
    </span>
  </div>
  <div>
    <span
      className="bid-list-card-title-post"
    >
      Location: 
    </span>
    Washington, DC
  </div>
  <div>
    <span
      className="bid-list-card-title-post"
    >
      Skill: 
    </span>
    CONSTRUCTION ENGINEERING (6218)
  </div>
  <div>
    <span
      className="bid-list-card-title-post"
    >
      Language: 
    </span>
    None
  </div>
  <div>
    <span
      className="bid-list-card-title-post"
    >
      Start date and End date: 
    </span>
    <StartEnd
      end="10/10/2018"
      start="10/10/2016"
    />
  </div>
</div>
`;
